include/GafferImageUI/ImageView.h

```diff
@@ -147,6 +147,8 @@ class GAFFERIMAGEUI_API ImageView : public GafferUI::View
 				Gaffer::PlugPtr createCounterpart( const std::string &name, Plug::Direction direction ) const override;
 		};
 
+		void setSoloChannel( int soloChannel );
```

Comment:
Views are intended to define all settings via plugs, _not_ via accessors and member data. The original design was such that ChannelChooser defined a `soloChannel` plug - this plug needs to remain as the only public interface (I presume this implies some refactoring).

Reply:
I'm not sure I understand enough of the concern to understand what refactoring is required.  I've switched the accessor from public to protected.  It makes sense that it's not part of the public interface, and is only to be called from ChannelChooser.  Is further refactoring required?

Reply:
The basic design of the View classes is that there's a central class which does the basics (ImageView in this case), and then "bolt on" classes that add in features (ChannelChooser in this case). Eventually I was thinking this might be exposed so that people could build more customised views (not everything makes sense to have on in the `gaffer view` app vs the `gaffer gui` app for instance). But in the meantime they're still a useful way of structuring the code, hopefully preventing it becoming a complete spaghetti. So I guess the question is whether responsibilities can remain separated, and if not, does ChannelChooser still make sense as a bolt-on component...